app/controllers/cart_items_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CartItemsController < ApplicationController
+  include CurrentCart
+  before_action :current_cart, only: %i[create destroy]
+  before_action :find_cart_item, only: %i[show destroy]
+
+  def new
+    @cart_item = CartItem.new
+  end
+
+  def create
+    menu_item = MenuItem.find(params[:menu_item_id])
+    @cart_item = @cart.add_menu_item(menu_item)
+
+    if @cart_item.save
+      render '/shared/cart_modal'
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    if @cart_item.quantity > 1
```

Comment:
can you explain why you need this?

Reply:
We have this cart modal with green and red buttons (I've attached the screenshot). If a quantity of the item is bigger than one and a user wants to reduce it, he or she clicks on a red button. We check the quantity of the item, if it is bigger than one then we reduce it by one via changing the quantity attribute of the item. After, we render the cart modal to reflect changes. But if the quantity is one and the user clicks on the red button, we delete the item from a table completely and render the cart. Also, we are checking if the cart is empty after deletion of the item if so we delete the cart as well and redirect the user to a restaurant page.
![Screenshot from 2021-08-29 09-42-47](https://user-images.githubusercontent.com/75825939/131241470-41fe648b-9ce8-43f2-9b8a-b878ae1db2c9.png)